### ISSUE TITLE
Deserialization problem with wrapped enumeration with tuple variant

### ISSUE THREAD

**Issue by bwsw** [rkyv/rkyv#389]
I encountered a problem with rkyv and was able to isolate it as follows:

```rust
#[cfg(test)]
mod tests {
    use rkyv::{Archive, Deserialize, Serialize};

    #[derive(Archive, Deserialize, Serialize, Debug, PartialEq, Clone)]
    #[archive(check_bytes)]
    pub enum Variant {
        Tensor((Vec<i64>, Vec<u8>)),
        Integer(i64),
        Float(f64),
    }

    #[derive(Archive, Deserialize, Serialize, Debug, PartialEq, Clone, Default)]
    #[archive(check_bytes)]
    pub struct Outer {
        values: Vec<Variant>,
    }

    #[test]
    fn test_ser_des_variant() {
        let mut outer = Outer::default();
        outer.values.push(Variant::Integer(1));
        outer.values.push(Variant::Float(2.0));
        outer
            .values
            .push(Variant::Tensor((vec![8, 3, 8, 8], [0; 3 * 8 * 8].to_vec())));

        let mut bytes = Vec::with_capacity(760);
        bytes.extend_from_slice(
            rkyv::to_bytes::<_, 756>(&outer)
                .expect("Failed to serialize VideoFrame")
                .as_ref(),
        );

        let o: Result<Outer, _> = rkyv::from_bytes(&bytes[..]);
        assert!(o.is_ok());
    }
}
```

Result:

```
thread 'test_rkyv_tuple::tests::test_ser_des_variant' panicked at 'assertion failed: o.is_ok()', src/test_rkyv_tuple.rs:36:9
```

Changing tuple to a simpler form solves the problem:

```
    #[derive(Archive, Deserialize, Serialize, Debug, PartialEq, Clone)]
    #[archive(check_bytes)]
    pub enum Variant {
        Tensor(Vec<i64>, Vec<u8>),
        Integer(i64),
        Float(f64),
    }
```

**Comment by djkoloski** [comment]
Minimized this a bit more and tracked it down to https://github.com/rkyv/bytecheck/commit/c5559cf59f3ee8232ca6b4be0f153de1e346eeeb. This issue is fixed with bytecheck 0.6.11, which rkyv uses as of 0.7.42 (now published).

**Comment by bwsw** [comment]
Checked. The bug is fixed.
